research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,346 @@
+import pandas as pd
+import urllib.request
+import os        
+import sys
+import zipfile
+import time
+
+def reporthook(blocknum, blocksize, totalsize):
+    readsofar = blocknum * blocksize
+    if totalsize > 0:
+        percent = readsofar * 1e2 / totalsize
+        s = "\r%5.1f%% %*d / %d" % (
+            percent, len(str(totalsize)), readsofar, totalsize)
+        sys.stderr.write(s)
+        if readsofar >= totalsize: # near the end
+            sys.stderr.write("\n")
+    else: # total size is unknown
+        sys.stderr.write("read %d\n" % (readsofar,))
+        
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_contas_2010.zip'
```

Comment:
Would you mind keep cols smaller than 80 chars? A clean way to have long strings in Python is:

```python
url = (
    'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas'
    '/prestacao_contas_2010.zip'
)
```

Reply:
Hi @cuducos, this is the most difficult part. I tried to keep it this way the whole code, but in many parts, the readability of the code would be compromised if I strictly followed this rule, what made me use # noqa in some parts of the code. This is my opinion, of course, and if you still think I should stick to this 80 chars rule, I will do that. Could you take a look in the new code?

Reply:
That's completely OK ; )